sdk/appconfiguration/Azure.Data.AppConfiguration/tests/FeatureFlagConfigurationSettingTests.cs

```diff
@@ -61,7 +111,19 @@ public void CanRountripValue(string value)
             var featureFlag = new FeatureFlagConfigurationSetting();
             featureFlag.Value = value;
 
-            Assert.AreEqual(value, featureFlag.Value);
+            try
+            {
+                using var expected = JsonDocument.Parse(value);
+                using var actual = JsonDocument.Parse(featureFlag.Value);
+
+                Assert.IsTrue(_jsonComparer.Equals(expected.RootElement, actual.RootElement));
+            }
+            catch (JsonException)
+            {
+                // For the cases that are not legal JSON, this exception will occur
+                // and we just want to make sure that the string value is set correctly.
+                Assert.AreEqual(value, featureFlag.Value);
```

Comment:
Why allow non-legal JSON?

Reply:
Because the service allows it.  The JSON schema for this setting type is a suggested convention, not a requirement.